commands/paymails.go

```diff
@@ -0,0 +1,13 @@
+package commands
+
+import "github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/querybuilders"
+
+// CreatePaymail defines the parameters required to create a new paymail address,
+// including associated metadata such as the public name and avatar.
+type CreatePaymail struct {
+	Metadata   querybuilders.Metadata `json:"metadata"`    // Metadata associated with the paymail as key-value pairs.
+	Key        string                 `json:"key"`         // The xpub key linked to the paymail.
+	Address    string                 `json:"address"`     // The paymail address to be created.
+	PublicName string                 `json:"public_name"` // The public display name associated with the paymail.
+	Avatar     string                 `json:"avatar"`      // The URL of the paymail's avatar image.
+}
```

Comment:
When does it "encoded from" or "decoded to" `json`?

Moreover, why you don't use models from `spv-wallet/models` ? They are intended to be used by `go-client`
https://github.com/bitcoin-sv/spv-wallet/blob/main/actions/admin/models.go#L11-L22

Reply:
> When does it "encoded from" or "decoded to" json?

Commands are used as request bodies.  To be more clear, this structures are not exactly pure command in _CQRS_ understanding. Setting the appropriate request bodies takes place individually in each API method call related to HTTP POST, PUT, UPDATE etc. 

> Moreover, why you don't use models from spv-wallet/models ? They are intended to be used by go-client
https://github.com/bitcoin-sv/spv-wallet/blob/main/actions/admin/models.go#L11-L22

This is a design issue on the SPV Wallet API side that should be solved first.  Current state of the models package doesn't include request bodies used across the APIs.

I would suggest to apply CQS (command query segregation) approach in SPV Wallet API. After this step all commands, queries can be easily replaced by go-cli -- what brings to the table less code, one consistent version of request/response models, clear separation and minimal imports.   
 
That's why a little bit of duplication was created as temporary solution that can be replaced very quickly.